frontend/server/src/ClaimRule.php

```diff
@@ -0,0 +1,60 @@
+<?php
+declare(strict_types=1);
+namespace \ParagonIE\Paseto\Rules;
+
+use ParagonIE\Paseto\{
+    JsonToken,
+    ValidationRuleInterface
+};
+use ParagonIE\Paseto\Exception\PasetoException;
+
+/**
+ * Class ClaimRule
+ * @package \ParagonIE\Paseto\Rules
+ */
+class ClaimRule implements ValidationRuleInterface {
+    /** @var string $failure */
+    protected $failure = 'OK';
+
+    /** @var string $rule */
+    protected $rule;
+
+    /** @var string $value */
+    protected $value;
+
+    /**
+     * ClaimRule constructor.
+     * @param string $audience
+     */
+    public function __construct(string $rule, string $value) {
+        $this->rule = $rule;
+        $this->value = $value;
+    }
+
+    /**
+     * @return string
+     */
+    public function getFailureMessage(): string {
+        return $this->failure;
+    }
+
+    /**
+     * @param JsonToken $token
+     * @return bool
+     */
+    public function isValid(JsonToken $token): bool {
+        try {
+            $claims = $token->getClaims();
+            if (!\hash_equals($this->value, $claims[$this->rule])) {
```

Comment:
```suggestion
            $value = $token->get($this->claim);
            if (!\hash_equals($this->value, $value)) {
```

es mejor sólo obtener el `$value` una sóla vez para usarlo también abajo.

Reply:
Listo